platform/android/java/lib/src/org/godotengine/godot/utils/PermissionsUtil.java

```diff
@@ -0,0 +1,143 @@
+package org.godotengine.godot.utils;
+
+import android.Manifest;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.PermissionInfo;
+import android.os.Build;
+import android.support.v4.content.ContextCompat;
+import java.util.ArrayList;
+import java.util.List;
+import org.godotengine.godot.Godot;
+
+/**
+ * This class has include Android permission operations. You should request permissions
+ * which has defined in manifest. With this util you can request permission which protection level is
+ * dangerous.
+ * @author Cagdas Caglak <cagdascaglak@gmail.com>
+ */
+public final class PermissionsUtil {
+
+	static final int REQUEST_RECORD_AUDIO_PERMISSION = 1;
+	static final int REQUEST_CAMERA_PERMISSION = 2;
+	static final int REQUEST_VIBRATE_PERMISSION = 3;
+	static final int REQUEST_ALL_PERMISSION_REQ_CODE = 1001;
+
+	private PermissionsUtil() {
+	}
+
+	/**
+	 * Request a dangerous permission. p_name must specified in <a href="https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/res/AndroidManifest.xml">this</a>
+	 * @param p_name the name of requested permission.
+	 * @param p_activity the caller activity name for this method.
+	 * @return true/false. "true" if permission was granted otherwise returns "false".
+	 */
+	public static boolean requestPermission(String p_name, Godot p_activity) {
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+			// Not necessary, asked on install already
+			return true;
+		}
+
+		if (p_name.equals("RECORD_AUDIO") && ContextCompat.checkSelfPermission(p_activity, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
+			p_activity.requestPermissions(new String[] { Manifest.permission.RECORD_AUDIO }, REQUEST_RECORD_AUDIO_PERMISSION);
+			return false;
+		}
+
+		if (p_name.equals("CAMERA") && ContextCompat.checkSelfPermission(p_activity, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+			p_activity.requestPermissions(new String[] { Manifest.permission.CAMERA }, REQUEST_CAMERA_PERMISSION);
+			return false;
+		}
+
+		if (p_name.equals("VIBRATE") && ContextCompat.checkSelfPermission(p_activity, Manifest.permission.VIBRATE) != PackageManager.PERMISSION_GRANTED) {
+			p_activity.requestPermissions(new String[] { Manifest.permission.VIBRATE }, REQUEST_VIBRATE_PERMISSION);
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Request dangerous permissions which have defined in AndroidManifest.xml. If some of selected
```

Comment:
The javadoc for this method should be:
 `Request dangerous permissions which are defined in the Android manifest file from the user.`

The second sentence is a repeat of the first one and can be dropped.